doc/tutorials/raspberry_electrophoresis/raspberry_electrophoresis.ipynb

```diff
@@ -257,66 +315,106 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "In order to make the colloid perfectly round, we now adjust the bead's positions to be exactly <tt>radius_col</tt> away\n",
-    "from the central bead."
+    "Now that the beads are arranged in the shape of a raspberry, the surface beads are made virtual particles\n",
+    "using the `VirtualSitesRelative` scheme. This converts the raspberry to a rigid body\n",
+    "in which the surface particles follow the translation and rotation of the central particle.\n",
+    "Newton's equations of motion are only integrated for the central particle.\n",
+    "It is given an appropriate mass and moment of inertia tensor (note that the inertia tensor\n",
+    "is given in the frame in which it is diagonal.)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
+   "cell_type": "markdown",
+   "metadata": {
+    "solution2": "hidden",
+    "solution2_first": true
+   },
    "source": [
-    "# this loop moves the surface beads such that they are once again exactly radius_col away from the center\n",
-    "# For the scalar distance, we use system.distance() which considers periodic boundaries\n",
-    "# and the minimum image convention\n",
-    "colPos = system.part[0].pos\n",
-    "for p in system.part[1:]:\n",
-    "    p.pos = (p.pos - colPos) / np.linalg.norm(system.distance(p, system.part[0])) * radius_col + colPos\n",
-    "    p.pos = (p.pos - colPos) / np.linalg.norm(p.pos - colPos) * radius_col + colPos"
```

Comment:
I took the liberty of moving this renormalization code into the minimization loop. Manually constraining the particles to lie on a sphere of radius `radius_col` after the minimization defeats the purpose of the minimization, as it makes the particles overlap and experience large forces. This can jeopardize the counter-ion minimization that follows later; in rare cases (depending on the numpy seed), my kernel would crash due to extreme torques:
```
INFO:root:Added 48 positive ions
INFO:root:Added 8 negative ions
INFO:root:Removing overlap between ions
python3: /work/jgrad/espresso-fork-PR/src/core/rotation.cpp:146: void propagate_omega_quat_particle(Particle&, double): Assertion `square >= 0.' failed.
[seraue:3021502] *** Process received signal ***
[seraue:3021502] Signal: Aborted (6)
[seraue:3021502] Signal code:  (-6)
```
Looking at the particle distribution in the visualizer, these crashes happen when the minimization wasn't long enough to uniformly distribute the particles on a sphere. The new algorithm periodically constrain the surface particles to be on a sphere, which makes it converge a lot faster and should result in a more uniform spherical distribution.